arch/x86/arch.c

```diff
@@ -55,6 +55,9 @@ uint8_t _kstack[PAGE_SIZE] __ALIGNED(sizeof(unsigned long));
 /* save a pointer to the multiboot information coming in from whoever called us */
 /* make sure it lives in .data to avoid it being wiped out by bss clearing */
 __SECTION(".data") uint32_t _multiboot_info;
+/* Try get the system table from tag 12, if present */
+/* `#define MULTIBOOT2_TAG_TYPE_EFI64    12` */
+__SECTION(".data") uint64_t _efi64_system_table;
```

Comment:
There shouldn't be a need to declare this in a special section since it's going to be overwritten by plain C code later.  The reason the _multiboot_info variable is declare this way is it's set in extremely early start.S code before it zeros out the BSS, so it has to be set aside so the BSS clearing doesn't wipe it out.